contracts/InsuranceArbitrator.sol

```diff
@@ -0,0 +1,153 @@
+    FinderInterface public finder;
+
+    /****************************************
+     *                EVENTS                *
+     ****************************************/
+
+    event policyIssued(
+        bytes32 indexed policyId,
+        address indexed insurer,
+        string insuredEvent,
+        address indexed insuredAddress,
+        address currency,
+        uint256 insuredAmount
+    );
+    event claimSubmitted(
+        uint256 claimTimestamp,
+        bytes32 indexed policyId,
+        address indexed insurer,
+        string insuredEvent,
+        address indexed insuredAddress,
+        address currency,
+        uint256 insuredAmount
+    );
+    event claimAccepted(
+        uint256 claimTimestamp,
+        bytes32 indexed policyId,
+        address indexed insurer,
+        string insuredEvent,
+        address indexed insuredAddress,
+        address currency,
+        uint256 insuredAmount
+    );
+    event claimRejected(
+        uint256 claimTimestamp,
+        bytes32 indexed policyId,
+        address indexed insurer,
+        string insuredEvent,
+        address indexed insuredAddress,
+        address currency,
+        uint256 insuredAmount
+    );
+
+    /**
+     * @notice Construct the InsuranceArbitrator
+     * @param _finderAddress DVM finder to find other UMA ecosystem contracts.
+     */
+    constructor(address _finderAddress) {}
+
+    /******************************************
+     *          INSURANCE FUNCTIONS           *
+     ******************************************/
+
+    /**
+     * @notice Deposits `insuredAmount` from the insurer and issues insurance policy to the insured beneficiary.
+     * @dev This contract must be approved to spend at least `insuredAmount` of `currency` token.
+     * @param insuredEvent short description of insured event. Potential verifiers should be able to evaluate whether
+     * this event had occurred as of claim time with binary yes/no answer.
+     * @param insuredAddress Beneficiary address eligible for insurance compensation.
+     * @param currency Denomination token for insurance coverage.
+     * @param insuredAmount Amount of insurance coverage.
+     * @return policyId Unique identifier of issued insurance policy.
+     */
+    function issueInsurance(
+        string calldata insuredEvent,
+        address insuredAddress,
+        address currency,
+        uint256 insuredAmount
+    ) external returns (bytes32 policyId) {}
+
+    /**
+     * @notice Anyone can submit insurance claim posting oracle bonding. Only one simultaneous claim per insurance
+     * policy is allowed.
+     * @dev This contract must be approved to spend at least `insuredAmount` * `oracleBondPercentage` + `finalFee` of
+     * `currency` token. This call requests and proposes that `insuredEvent` had ocured through Optimistic Oracle.
+     * @param policyId Identifier of claimed insurance policy.
+     */
+    function submitClaim(bytes32 policyId) external {}
```

Comment:
so to make sure I understand the next steps: you'll query the OO directly and place the burden of all settlement logic for the request within the OO? so this contract wont have any dispute logic or anything related, right?

Reply:
Yes, dispute and settlement is made directly on OO. Upon settlement OO callbacks this contract and handles the claim logic.